src/v/security/audit/schemas/application_activity.h

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2023 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "security/audit/schemas/schemas.h"
+#include "security/audit/schemas/types.h"
+
+namespace security::audit {
+class api_activity final : public ocsf_base_event {
+public:
+    enum class activity_id : uint8_t {
+        unknown = 0,
+        create = 1,
+        read = 2,
+        update = 3,
+        delete_id = 4,
+        other = 99
+    };
+
+    enum class status_id : uint8_t {
+        unknown = 0,
+        success = 1,
+        failure = 2,
+        other = 99
+    };
+
+    api_activity(
+      activity_id activity_id,
+      actor&& actor,
+      api&& api,
+      network_endpoint&& dst_endpoint,
+      std::optional<http_request>&& http_request,
+      std::vector<resource_detail>&& resources,
+      severity_id severity_id,
+      network_endpoint&& src_endpoint,
+      status_id status_id,
+      timestamp_t time,
+      api_activity_unmapped&& unmapped)
+      : ocsf_base_event(
+        category_uid::application_activity,
+        class_uid::api_activity,
+        severity_id,
+        time,
+        activity_id)
+      , _activity_id(activity_id)
+      , _actor(std::move(actor))
+      , _api(std::move(api))
+      , _dst_endpoint(std::move(dst_endpoint))
+      , _http_request(std::move(http_request))
+      , _resources(std::move(resources))
+      , _src_endpoint(std::move(src_endpoint))
+      , _status_id(status_id)
+      , _unmapped(std::move(unmapped)) {}
+
+    auto equality_fields() const {
+        return std::tie(
+          _activity_id,
+          _actor,
+          _api,
+          _dst_endpoint.addr.host(),
+          _http_request,
+          _resources,
+          _src_endpoint.addr.host(),
```

Comment:
do these .host() calls cause a reference to a temporary to be returned?

Reply:
They do, but `std::tie` returns a `std::tuple` of lvalue references, so I _think_ it's ok?  Or does it turn it to a reference of a reference?

Reply:
i think it depends on what host() returns and the the lifetime of the thing it returns. here is a use-after-free demo

https://godbolt.org/z/j9TzWTEWa

Reply:
`host()` returns reference to a member variable of `unresolved_address`.  That's held by the `_dest/_src_endpoint`

Reply:
got it. that's what i was looking for. it's always a bit scary when those interfaces could change and return a value causing a use-after-free bug